File: utils/augment.py
from .models import GameMoves, AugmentedGameMoves, MoveOptions, AugmentedMoveOptions
from .path_finding import shortest_path

def _augment_move(game_state: str, move: GameMoves) -> AugmentedGameMoves:
    is_valid = True
    for m in move.moves:
        path = shortest_path(game_state, (m.x, m.y), m.last_move)
        if len(path) == 0:
            is_valid = False
        break

    return AugmentedGameMoves(
        moves=move.moves,
        goal=move.goal,
        is_valid=is_valid
    )

def augment_game_moves(game_state: str, moves: MoveOptions) -> AugmentedMoveOptions:
    augmented_moves = [_augment_move(game_state, m) for m in moves.options]
    return AugmentedMoveOptions(options=augmented_moves)

File: .opencode/tools/utils/augment.ts
import { GameMoves, AugmentedGameMoves, MoveOptions, AugmentedMoveOptions, type Position } from "./models.js";
import { shortestPath } from "./path_finding.js";

function augmentMove(gameState: string, move: GameMoves): AugmentedGameMoves {
  let isValid = true;

  for (const m of move.moves) {
    const path = shortestPath(gameState, [m.x, m.y], m.last_move);
    if (path.length === 0) {
      isValid = false;
      break;
    }
  }

  return {
    moves: move.moves,
    goal: move.goal,
    is_valid: isValid,
  };
}

export function augmentGameMoves(
  gameState: string,
  moves: MoveOptions
): AugmentedMoveOptions {
  const augmentedMoves = moves.options.map((m) => augmentMove(gameState, m));
  return { options: augmentedMoves };
}
